samza-api/src/main/java/org/apache/samza/config/ConfigLoader.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.config;
+
+import java.util.Map;
+
+
+/**
+ * Build a {@link org.apache.samza.config.Config}
+ */
+public interface ConfigLoader {
```

Comment:
Can we add more descriptive java docs? 
1. It will be helpful if you can talk about how and when this interface is used.
2. What if `ConfigLoader` implementors want to handle dependency injection? Do we need to provide a factory for the config loader implementors?

Reply:
Added more descriptions.